stock_orderpoint_generator/models/orderpoint_template.py

```diff
@@ -42,32 +86,88 @@ class OrderpointTemplate(models.Model):
+            'auto_min_date_end', 'auto_max_date_start', 'auto_max_date_end',
+            'auto_max_qty_criteria', 'auto_max_qty',
+        ]
+
     def _disable_old_instances(self, product_ids):
-        """ Clean old instance by setting those inactives
-        """
+        """Clean old instance by setting those inactives"""
         orderpoints = self.env['stock.warehouse.orderpoint'].search(
-            [('product_id', 'in', product_ids)]
+            [('product_id', 'in', product_ids.ids)]
         )
         orderpoints.write({'active': False})
 
+    @api.model
+    def _get_product_qty_by_criteria(
+            self, product_ids, location_id, from_date, to_date, criteria):
+        """Returns a dict with product ids as keys and the resulting
+           calculation of historic moves according to criteria"""
+        stock_qty_history = product_ids._compute_historic_quantities_dict(
+            location_id=location_id,
+            from_date=from_date,
+            to_date=to_date)
+        criteria_list = [x[0] for x in self._get_auto_criteria_selection()]
+        if criteria not in criteria_list:
+            return {x: 0 for x, y in stock_qty_history.items()}
+        if criteria == 'max':
+            return {x: max(y['stock_history'])
+                    for x, y in stock_qty_history.items()}
+        if criteria == 'median':
+            return {x: median_high(y['stock_history'])
+                    for x, y in stock_qty_history.items()}
+        if criteria == 'avg':
+            return {x: mean(y['stock_history'])
+                    for x, y in stock_qty_history.items()}
+        if criteria == 'min':
+            return {x: min(y['stock_history'])
+                    for x, y in stock_qty_history.items()}
+
     def _create_instances(self, product_ids):
-        """ Create instances of model using template inherited model
-        """
+        """Create instances of model using template inherited model and
+           compute autovalues if needed"""
         orderpoint_model = self.env['stock.warehouse.orderpoint']
         for record in self:
             for data in record.copy_data():
-                data.pop('auto_generate')
-                data.pop('auto_product_ids')
-                data.pop('auto_last_generation')
+                for discard_field in self._template_fields_to_discard():
+                    data.pop(discard_field)
+                if record.auto_min_qty:
+                    stock_min_qty = (
+                        self._get_product_qty_by_criteria(
+                            product_ids,
+                            location_id=record.location_id,
+                            from_date=record.auto_min_date_start,
+                            to_date=record.auto_min_date_end,
+                            criteria=record.auto_min_qty_criteria,
+                        ))
+                if record.auto_max_qty:
+                    stock_max_qty = (
```

Comment:
Check if the criteria is the same as minimum one, and if so, avoid again the computation.